psl/psl-core/src/validate/validation_pipeline/validations.rs

```diff
@@ -123,7 +123,7 @@ pub(super) fn validate(ctx: &mut Context<'_>) {
             indexes::supports_clustering_setting(index, ctx);
             indexes::clustering_can_be_defined_only_once(index, ctx);
             indexes::opclasses_are_not_allowed_with_other_than_normal_indices(index, ctx);
-            indexes::composite_types_are_not_allowed_in_index(index, ctx);
+            indexes::composite_type_in_compound_unique_index(index, ctx);
```

Comment:
What about normal indexes? Are those supported with a mix of scalar and composite fields?

Reply:
They don't actually hit this error (fulltext also doesn't). Not to say they're _supported_ but they run into different issues that are unrelated to the original crash report. Only `@@unique` causes a panic from what I've found